### DISCUSSION TITLE
File size limits on the new repository

### DISCUSSION THREAD

**Discussion by bjr70** [MESH-Research/knowledge-commons-repository#107] [General]
The current repository has a size limit of 100MB and allows only one file per upload. Invenio allows the upload of multiple files per deposit. We know that we can use S3 to store files, so that opens us up to increasing that file size limit, but by how much and by deposit or per file?

- If a deposit has more than one file does the size limit apply to the deposit as a whole, or to each individual file?
- Should there be a limit? If so should it be 500GB, 1GB, more? 

Are there other things we need to consider? Drop them below.

**Comment by monotasker** [comment]
If we make the limit per-file (rather than per-deposit) then will we need to limit the number of files that can be attached to an upload? Otherwise someone might simply split a large video file (say) into several smaller chunks. So maybe making the limit apply to the whole deposit makes sense?

If we do that, then maybe it would make the most sense to frame it for users as a "storage limit" or "storage space" for each deposit. Then maybe in the file upload widget display the remaining storage space for the deposit? That would prevent users from having to mentally add up the sizes of their files.

**Comment by bjr70** [comment]
@mikethicke @tzouris @rekmla any thoughts on this?

**Comment by tzouris** [comment]
Individual S3 objects can be up to 5TB in size, but the largest object that can be uploaded in a single PUT is 5 GB, so let's think of that as our technical limit. Being able to upload multiple files is important. For instance, you can upload a PDF and a separate spreadsheet with data tables. I don't know if we should impose a limit to the number of files. I like Ian's idea of an overall deposit limit though. And maybe we could set it to 5 GB (even though each file will be a separate PUT call) in order to allow for multimedia and VR content as well? If we start getting requests for larger deposits, we can revisit.

**Comment by mikethicke** [comment]
I think a 5 GB per-deposit cap is reasonable. However:

- I would check both file size and total size. This will give us more flexibility in the future if we want to raise the limit for total size, but still face the 5 GB limit imposed by S3.
- We might consider per-user and per-deposit flags that are settable by administrators that allow particular users or deposits to bypass the per-deposit limit. I would probably create and check for the flags now, even if the UI doesn't exist to use them yet.
